src/main/java/frc/robot/subsystems/drivetrain/commands/JoystickDrive.java

```diff
@@ -27,27 +28,31 @@ public JoystickDrive(SwerveDrive swerveDrive, Joystick leftJoystick, Joystick ri
 
     @Override
     public void execute() {
-        double vx = -leftJoystick.getY();
-        double vy = -leftJoystick.getX();
-        double omega = -rightJoystick.getX();
-
-        vx = forwardLimiter.calculate(vx);
-        vy = strafeLimiter.calculate(vy);
-        omega = rotationLimiter.calculate(omega);
-
-        double magnitude = Math.hypot(vx, vy);
-        double angle = Math.atan2(vy, vx);
-        magnitude = MathUtil.applyDeadband(magnitude, 0.2);
-        vx = Math.cos(angle) * magnitude;
-        vy = Math.sin(angle) * magnitude;
-        omega = MathUtil.applyDeadband(omega, 0.2);
-
-        DriveSignal signal = new DriveSignal(
-                vx * SwerveConstants.MAX_VELOCITY_METERS_PER_SECOND,
-                vy * SwerveConstants.MAX_VELOCITY_METERS_PER_SECOND,
-                omega * SwerveConstants.MAX_ANGULAR_VELOCITY_RADIANS_PER_SECOND,
-                new Translation2d(),
-                !leftJoystick.getTop());
-        swerveDrive.drive(signal);
+        if (leftJoystick.getRawButton(Ports.UI.JOYSTICK_TOP_RIGHT_BUTTON)) {
+            swerveDrive.lock();
+        } else {
+            double vx = -leftJoystick.getY();
+            double vy = -leftJoystick.getX();
+            double omega = -rightJoystick.getX();
+
+            vx = forwardLimiter.calculate(vx);
+            vy = strafeLimiter.calculate(vy);
+            omega = rotationLimiter.calculate(omega);
+
+            double magnitude = Math.hypot(vx, vy);
+            double angle = Math.atan2(vy, vx);
+            magnitude = MathUtil.applyDeadband(magnitude, 0.1);
+            vx = Math.cos(angle) * magnitude;
+            vy = Math.sin(angle) * magnitude;
+            omega = MathUtil.applyDeadband(omega, 0.1);
+
+            DriveSignal signal = new DriveSignal(
+                    vx * SwerveConstants.MAX_VELOCITY_METERS_PER_SECOND,
+                    vy * SwerveConstants.MAX_VELOCITY_METERS_PER_SECOND,
+                    omega * SwerveConstants.MAX_ANGULAR_VELOCITY_RADIANS_PER_SECOND,
+                    new Translation2d(),
+                    !leftJoystick.getTop());
+            swerveDrive.drive(signal);
+        }
```

Comment:
It's better to reduce indentation when possible. By putting a `return` statement inside the `if`, you can delete the `else` clause:
```suggestion
                   if (leftJoystick.getRawButton(Ports.UI.JOYSTICK_TOP_RIGHT_BUTTON)) {
            swerveDrive.lock();
            return;
        } 
        double vx = -leftJoystick.getY();
        double vy = -leftJoystick.getX();
        double omega = -rightJoystick.getX();

        vx = forwardLimiter.calculate(vx);
        vy = strafeLimiter.calculate(vy);
        omega = rotationLimiter.calculate(omega);

        double magnitude = Math.hypot(vx, vy);
        double angle = Math.atan2(vy, vx);
        magnitude = MathUtil.applyDeadband(magnitude, 0.1);
        vx = Math.cos(angle) * magnitude;
        vy = Math.sin(angle) * magnitude;
        omega = MathUtil.applyDeadband(omega, 0.1);

        DriveSignal signal = new DriveSignal(
                vx * SwerveConstants.MAX_VELOCITY_METERS_PER_SECOND,
                vy * SwerveConstants.MAX_VELOCITY_METERS_PER_SECOND,
                omega * SwerveConstants.MAX_ANGULAR_VELOCITY_RADIANS_PER_SECOND,
                new Translation2d(),
                !leftJoystick.getTop());
        swerveDrive.drive(signal);
```